SpaceHosting.Index.Tests/Sparnn/Distances/JaccardMethodsComparisonTests.cs

```diff
@@ -0,0 +1,85 @@
+        private SparnnIndex jaccardBinarySingleOrientedIndex = null!;
+        private SparseVector[] vectorsToSearch = null!;
+
+        private const int VectorSpaceSize = 15_000;
+        private const int VectorSize = 15_000;
+        private const int MinimumValuable = 5;
+        private const int MaximumValuable = 15;
+        private const int FeatureVectorsCount = 1;
+
+        const int indicesNumber = 2;
+        const int clusterSize = 1000;
+        private static Random Random = null!;
+
+
+        private static IEnumerable<Tuple<int, double>> GetIndexedValues(int count)
+        {
+            return Enumerable
+                .Range(0, VectorSize)
+                .RandomSubset(count, Random)
+                .Select(i => new Tuple<int, double>(i, 1.0));
+        }
+
+        private static IEnumerable<SparseVector> GenerateVectors(int count)
+        {
+            return Enumerable.Range(0, count)
+                .Select(_ => Random.Next(MinimumValuable, MaximumValuable))
+                .Select(valuableCount => MSparseVector.OfIndexedEnumerable(VectorSize, GetIndexedValues(valuableCount)).ToModelVector());
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            Random = new Random(42);
+            jaccardBinaryIndex = new SparnnIndex(new MatrixMetricSearchSpaceFactory(MatrixMetricSearchSpaceAlgorithm.JaccardBinary), indicesNumber, clusterSize, VectorSize);
+            jaccardBinarySingleOrientedIndex = new SparnnIndex(new MatrixMetricSearchSpaceFactory(MatrixMetricSearchSpaceAlgorithm.JaccardBinarySingleFeatureOriented), indicesNumber, clusterSize, VectorSize);
+            
+            var baseVectors = GenerateVectors(VectorSpaceSize).Select(x => ((long)Random.Next(), x));
+
+            foreach (var batch in baseVectors.Batch(size: 1000, b => b.ToArray()))
+            {
+                jaccardBinaryIndex.AddBatch(batch);
+                jaccardBinarySingleOrientedIndex.AddBatch(batch);
+            }
+
+            vectorsToSearch = GenerateVectors(FeatureVectorsCount).ToArray();
+        }
+
+        [TestCase(1)]
+        [TestCase(10)]
+        [TestCase(100)]
+        [TestCase(1000)]
+
+        public void FindNearestReturnsSameResult(int kNearestCount)
+        {
+            double AccumulateNearestVectorResultDistanse(SparnnIndex sparnnIndex)
+            {
+                var searchResult = sparnnIndex.FindNearest(vectorsToSearch, kNearestCount);
+                searchResult.Should().HaveCount(1);
+                return searchResult[0].Select(x => x.Distance).Sum();
+            }
+
+            var firstMethodResultIds = AccumulateNearestVectorResultDistanse(jaccardBinaryIndex);
+            var secondMethodResultIds = AccumulateNearestVectorResultDistanse(jaccardBinarySingleOrientedIndex);
+            var diff = Math.Abs(firstMethodResultIds - secondMethodResultIds);
+            const double oneVectorMaxAbsoluteError = 1.0;
```

Comment:
I would suggest adding the ability to fixate random number generator which used in `NonTerminalClusterIndex` [here](https://github.com/kontur-model-ops/space-hosting-index/blob/7229453d061af476f5670c62f1d5749352ddcdc3/SpaceHosting.Index/Sparnn/Clusters/NonTerminalClusterIndex.cs#L127) in order to make these tests deterministic.

Reply:
> Would you please post benchmark results here to show perf improvements?

![image](https://user-images.githubusercontent.com/10279188/122908969-e8341700-d37e-11eb-8dd3-a2417bc9c168.png)

Reply:
https://github.com/kontur-model-ops/space-hosting-index/pull/1/commits/5a197760b6d3fc871c6de9ead0269170e3603366
fixed.
Note that its is usefull generally, but not in that case: mistake was in tests, vectors with non-unique id were generated; it fixes and both methods do return same sum of distances now